### ISSUE TITLE
[Bindings] `TwoWaySeq`

### ISSUE THREAD

**Issue by xperiandri** [elmish/Elmish.WPF#610]
**High-level description**
`TokenizingTextBox` from WCT modifies the collection supplied to `ItemsSource`. Hence changes must be propagated back to the model.

I've implemented `TwoWaySeq` binding in my fork
See commits in https://github.com/Ecierge/Elmish.Uno/commits/eCierge/

Backport if you consider it useful

**Comment by YkTru** [comment]
Doesnt "NotifyCollectionChangedEventArgs" bind you to WPF (I know its not necessary to use WPF to use it, though I guess the intention is to communicate *directly* with your WPF project?)

**Comment by YkTru** [comment]
Doesnt 'NotifyCollectionChangedEventArgs' bind you to .Net? I mean, can't notification be done with Msg + updates + event-to-command (if you need eventArgs) behaviors (https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/behaviors/event-to-command-behavior)?

**Comment by YkTru** [comment]
I may be confused, so please correct me if I'm wrong, but my intuition is as follows:

1- if the library is to be linked to dotNet (which is inevitable for UNO/WPF), all dotNet functionality must be “hidden” (as in this [file](https://github.com/elmish/Elmish.WPF/blob/master/src/Elmish.WPF/ViewModels.fs)), i.e. the MVU part must not use anything from .Net (with the exception of “Window” in “main”)

2- if I need to get data from the user interface and “Mode=TwoWay” isn't enough, I use Behaviors if it's about events or any other situation (personally, I use Behaviors a LOT to keep MVU independent and clean).

3- otherwise, I have an F# project called “[ProjectName].WPF.Fs” which depends on “[ProjectName].Core” (parallel to C# "[ProjectName].WPF") in which I do all my "WPF dependant" code; I used it recently to do some clean DataTemplateSelector code (for a Form Designer), and I think I'll do my behaviors in F# too (I really prefer the synthax + error handling).

Am I misunderstanding something?

**Comment by xperiandri** [comment]
Well, I do not argue that the API may be better than I implemented.
The issue is that if an `ObservableCollection` is modified from UI you will never get such update. So we need to get it somehow

**Comment by YkTru** [comment]
@xperiandri Well, too bad our efforts to spark conversation aren't taken into account. I hope you'll continue to be interested in ElmishWPF/UNO and use it, otherwise I'll be very lonely..

Thank you for sharing your code and insights.

**Comment by xperiandri** [comment]
Maybe you become a maintainer of this repo? @YkTru

**Comment by YkTru** [comment]
Haha! I was thinking more about you!

**Comment by marner2** [comment]
So the issue here is to support hooking up a UI-supplied `ObservableCollection` of items? I'm pretty sure (at least WPF) XAML controls are not supposed to ever do that. What's the reasoning behind it?

**Comment by xperiandri** [comment]
Yes. As I specified I found that on the TokenizingTextBox control
